app/services/search_item_res.rb

```diff
@@ -40,6 +40,29 @@ def combine_highlights
     end
   end
 
+  def find_source_from_top_hits(top_hits, field, key)
+    # elasticsearch stores nested source results without the "path"
+    nested_child = field.split(".").last
+    hit = top_hits.first.dig("_source", nested_child)
+    # if this is a multivalued field (for example: works or places),
+    # ALL of the values come back as the source, but we only want
+    # the single value from which the key was derived
+    if hit.class == Array
+      # I don't love this, because we will have to match exactly the logic
+      # that got us the key to get this to work
+      match_index = hit
+        .map { |s| remove_nonword_chars(s) }
+        .index(remove_nonword_chars(key))
+      matches = hit.map { |s| remove_nonword_chars(s) }
+      # if nothing matches the original key, return the entire source hit
+      # should return a string, regardless
+      return match_index ? hit[match_index] : hit.join(" ")
```

Comment:
I don't see `matches` used after its assignment. Is the end of the last line I've highlighted here a typo? Should it end with `matches.join(" ")`?

Reply:
This is a good question. I will look into it but I bet you're right. It was once working and then I renamed variables and things trying to make it more readable but probably broke it instead :P